src/Sarif.Viewer.VisualStudio/ResultTextMarker.cs

```diff
@@ -72,48 +126,169 @@ public Region Region
+        /// <param name="resultId">The identifier of the <see cref="SarifErrorListItem"/> that this marker belongs to.</param>
+        /// <param name="region">The original source region from the SARIF log file.</param>
+        /// <param name="fullFilePath">The full file path of the location in the SARIF result.</param>
+        /// <param name="color">The non-highlighted color of the marker.</param>
+        /// <param name="highlightedColor">The highlighted color of the marker.</param>
+        /// <param name="errorType">The error type as defined by <see cref="Microsoft.VisualStudio.Text.Adornments.PredefinedErrorTypeNames"/>.</param>
+        /// <param name="tooltipContent">The tool tip content to display in Visual studio.</param>
+        /// <param name="context">The data context for this result marker.</param>
+        /// <remarks>
+        /// The tool tip content could be as simple as just a string, or something more complex like a WPF/XAML object.
+        /// </remarks>
+        public ResultTextMarker(int runIndex, int resultId, Region region, string fullFilePath, string color, string highlightedColor, string errorType, object tooltipContent, object context)
         {
-            this.runIndex = runIndex;
+            this.ResultID = resultId;
+            this.RunIndex = runIndex;
+            this.FullFilePath = fullFilePath;
             this.region = region ?? throw new ArgumentNullException(nameof(region));
-            FullFilePath = fullFilePath;
-            Color = DEFAULT_SELECTION_COLOR;
+            this.Color = color;
+            this.HighlightedColor = highlightedColor;
+            this.ToolTipeContent = tooltipContent;
+            this.ErrorType = errorType;
+            this.Context = context;
+        }
+
+        public IEnumerable<ISarifLocationTag> GetTags<T>(ITextBuffer textBuffer, IPersistentSpanFactory persistentSpanFactory)
+            where T : ITag
+        {
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            List<ISarifLocationTag> tags = new List<ISarifLocationTag>();
+
+            if (this.persistentSpan == null)
+            {
+                if (!this.TryToFullyPopulateRegion())
+                {
+                    return tags;
+                }
+
+                if (!TryCreateTextSpanWithinDocumentFromSourceRegion(this.fullyPopulatedRegion, textBuffer, out TextSpan documentSpan))
+                {
+                    return tags;
+                }
+
+                this.persistentSpan = persistentSpanFactory.Create(
+                            textBuffer.CurrentSnapshot,
+                            startLine: documentSpan.iStartLine,
+                            startIndex: documentSpan.iStartIndex,
+                            endLine: documentSpan.iEndLine,
+                            endIndex: documentSpan.iEndIndex,
+                            trackingMode: SpanTrackingMode.EdgeInclusive);
+            }
```

Comment:
This needs to be factored out to a different function.

Reply:
Move this to this.TryCreatePersistentSpan()

---
In reply to: [501927601](https://github.com/microsoft/sarif-visualstudio-extension/pull/214#discussion_r501927601) [](ancestors = 501927601)